eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelTypes.scala

```diff
@@ -173,9 +173,24 @@ sealed trait HasCommitments extends Data {
 
 case class ClosingTxProposed(unsignedTx: Transaction, localClosingSigned: ClosingSigned)
 
-case class LocalCommitPublished(commitTx: Transaction, claimMainDelayedOutputTx: Option[Transaction], htlcSuccessTxs: List[Transaction], htlcTimeoutTxs: List[Transaction], claimHtlcDelayedTxs: List[Transaction], irrevocablySpent: Map[OutPoint, ByteVector32])
-case class RemoteCommitPublished(commitTx: Transaction, claimMainOutputTx: Option[Transaction], claimHtlcSuccessTxs: List[Transaction], claimHtlcTimeoutTxs: List[Transaction], irrevocablySpent: Map[OutPoint, ByteVector32])
-case class RevokedCommitPublished(commitTx: Transaction, claimMainOutputTx: Option[Transaction], mainPenaltyTx: Option[Transaction], htlcPenaltyTxs: List[Transaction], claimHtlcDelayedPenaltyTxs: List[Transaction], irrevocablySpent: Map[OutPoint, ByteVector32])
+case class LocalCommitPublished(commitTx: Transaction, claimMainDelayedOutputTx: Option[Transaction], htlcSuccessTxs: List[Transaction], htlcTimeoutTxs: List[Transaction], claimHtlcDelayedTxs: List[Transaction], irrevocablySpent: Map[OutPoint, ByteVector32]) {
```

Comment:
I'm not a fan of this change, but don't have a better alternative.

Reply:
I can put that check only in `isClosingTypeAlreadyKnown` if you prefer (instead of adding this field to those case classes)?

Reply:
Is there an issue with our model (should that `irrevocablySpent` be at the upper level)?

Reply:
> How about just testing irrevocablySpent.nonEmpty?

I think it's dangerous, it feels safer to verify explicitly the txs we built. 
If in the future we start adding other confirmed txs in there that aren't necessarily for the txs we built, we'd create a nasty regression.

> should that irrevocablySpent be at the upper level

I don't know, I don't think it's needed right now to move it to an upper level. I found a small issue for revoked commitments with the latest commit, which I'm fixing. I think that moving this to have it only contained inside `isClosingTypeAlreadyKnown` is the right way to fix the issue at hand. It doesn't spill to other use-cases AFAICT, so no need to break the current model apart.